src/rust/engine/src/externs.rs

```diff
@@ -488,51 +500,88 @@ where
   output
 }
 
```

Comment:
Does our logging code justify a new module at this point?

Reply:
I initially had this in a new module, but doing so required me to make either `with_externs` public, or make a public wrapper around `e.log`, both of which made me a bit sad... Any ideas/thoughts?

Reply:
Yea, fine with me. Not sure what the way forward on this will be... other than finding a rule model we like and then pruning the `externs` to fit it.